text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,567 @@
+      'callingcode': new NumberAttr({ min: 1, max: 3 }),
+    },
+  }
+});
+```
+
+*Note* that, custom attributes cannot be marked as required.
+
+*Note*: By default, the standard attributes 'email' and/or 'phone\_number' will automatically be marked required if
+they are one of the verified contact methods. See [Sign up](#sign-up) for details on verified contact methods.
+
+Additionally, two properties `mutable` and `adminOnly` properties can be set for each custom attribute. The former
+specifies that the property can be modified by the user while the latter specifies that it can only be modified by the
+app's administrator and not by the user (using their access token).
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * No standard attributes are marked required.
+> * For all custom attributes, `mutable`: true & `adminOnly`: false.
+
+> Internal note: Implemented via UserPool-SchemaAttribute
+
+> Internal note: Follow up - is mutable = false and adminOnly = true allowed?
+
+### Security
+
+User pools can be configured to enable MFA. It can either be turned off, set to optional or made required. Setting MFA
+to optional means that individual users can choose to enable it. Phone numbers must be verified if MFA is enabled.
+Additionally, MFA can be sent either via SMS text message or via a time-based software token.
+See the [documentation on MFA](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-mfa.html) to
+learn more.
+
+This can be configured by setting the `mfa.enforcement` option under `security` properties to be one of the values in
+the `MfaEnforcement` enum. Available values are `REQUIRED`, `OPTIONAL`, `OFF`.
+The type of MFA can be configured by its peer property `type` which can be set to a list of values in the enum
+`MfaType`. The available values are `SMS` and `SOFTWARE_TOKEN`.
+
+User pools can be specify the constraints that should be applied when users choose their password. Constraints such as
+minimum length, whether lowercase, numbers and/or symbols are required can be specified.
+
+In order to send an SMS, Cognito needs an IAM role that it can assume with permissions that allow it to send an SMS on
+behalf of the AWS account. By default, CDK will create this IAM user but allows for it to be overridden via the
+`smsRole` permissions.
+
+> These are the defaults that will be part of tsdoc, and not part of the README -
+> * security.mfa.enforcement: OPTIONAL
+> * security.mfa.type: SMS
+> * security.passwordPolicy.minLength: 8
+> * security.passwordPolicy.required - lowercase, numbers and symbols
+> * security.smsRole - assumable by `cognito-idp.amazonaws.com` and permitting `sns:Publish` action against resource `*`.
+>   The role assumption will be conditioned on a strict equals on an ExternalId that will be unique to this user pool.
+
+Code sample:
+
+```ts
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  security: {
+    mfa: {
```

Comment:
I think I'd like to encourage a factory function here again, and same for `passwordPolicy`. Or flatten out the properties into the top-level props object.

We have a construct library design guideline to flatten out the properties namespace.